src/components/Personalization/handlers/createProcessPropositions.js

```diff
@@ -10,6 +10,13 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
+const executeSequentially = promiseFunctions =>
+  promiseFunctions.reduce((sequence, promiseFunction) => {
+    return sequence.then(result =>
+      promiseFunction().then(nextResult => [...result, nextResult])
```

Comment:
If you change this line, then you don't have to make sure everything is a promise:
```
return sequence.then(result => promiseFunction()).then(nextResult => [...result, nextResult]);
```
.then() always returns a promise

Reply:
This code is invalid, since `result` is out of scope in the second `then` clause.
```js
return sequence
  .then((result) => promiseFunction())
  .then((nextResult) => [...result, nextResult]);
```

what problem are you trying to solve?